lab_3/main.py

```diff
@@ -20,8 +21,41 @@ def tokenize_by_sentence(text: str) -> tuple:
          (('_', 'h', 'e', '_'), ('_', 'i', 's', '_'), ('_', 'h', 'a', 'p', 'p', 'y', '_'))
          )
     """
-    pass
-
+    if not isinstance(text, str):
+        return ()
+    invaluable_trash = ['`', '~', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '-', '+',
+                        '=', '{', '[', ']', '}', '|', '\\', ':', ';', '"', "'", '<', ',', '>',
+                        '/', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']
+    text = text.lower()
+    for symbol in invaluable_trash:
+        text = text.replace(symbol, '')
+    for letter in text:
+        if letter == 'ö':
+            text = text.replace(letter, 'oe')
+        elif letter == 'ü':
+            text = text.replace(letter, 'ue')
+        elif letter == 'ä':
+            text = text.replace(letter, 'ae')
+        elif letter == 'ß':
+            text = text.replace(letter, 'ss')
+    regexp = re.compile('[.!?] ?')
+    sentences = re.split(regexp, text)
+    cleaned_sentences = []
+    for sentence in sentences:
+        cleaned_sentence = sentence.strip()
+        if cleaned_sentence:
+            cleaned_sentences.append([cleaned_sentence])
+    for ind, sentence in enumerate(cleaned_sentences):
```

Comment:
ideally, you can replace all these for loops with just one for loop

Reply:
is it necessary to change it or can I leave it as it is?